src/sentry/integrations/slack/webhooks/event.py

```diff
@@ -249,15 +255,23 @@ def on_link_shared(self, request: Request, slack_request: SlackDMRequest) -> boo
             "unfurls": orjson.dumps(results).decode(),
         }
 
-        client = SlackSdkClient(integration_id=slack_request.integration.id)
-        try:
-            client.chat_unfurl(
-                channel=data["channel"],
-                ts=data["message_ts"],
-                unfurls=payload["unfurls"],
-            )
-        except SlackApiError:
-            _logger.exception("on_link_shared.unfurl-error", extra=logger_params)
+        with MessagingInteractionEvent(
+            interaction_type=MessagingInteractionType.UNFURL_LINK,
+            spec=SlackMessagingSpec(),
+        ).capture() as lifecycle:
+            client = SlackSdkClient(integration_id=slack_request.integration.id)
+            try:
+                client.chat_unfurl(
+                    channel=data["channel"],
+                    ts=data["message_ts"],
+                    unfurls=payload["unfurls"],
+                )
+            except SlackApiError as e:
+                lifecycle.add_extras(logger_params)
+                if organization_id == SUPERUSER_ORG_ID:
+                    lifecycle.add_extra("unfurls", payload["unfurls"])
```

Comment:
can you tell me what the flow is to get to this case? Like do we only unfurl if someone is superusering?

Reply:
We're only going to log the blocks that failed to unfurl for the Sentry org